parser/prism/Translator.cc

```diff
@@ -1803,11 +1803,40 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
             auto rescueModifierNode = down_cast<pm_rescue_modifier_node>(node);
             auto body = translate(rescueModifierNode->expression);
             auto rescue = translate(rescueModifierNode->rescue_expression);
+
+            if (!directlyDesugar || !hasExpr(body, rescue)) {
+                NodeVec cases;
+                // In rescue modifiers, users can't specify exceptions and the variable name so they're null
+                cases.emplace_back(make_unique<parser::Resbody>(location, nullptr, nullptr, move(rescue)));
+                return make_unique<parser::Rescue>(location, move(body), move(cases), nullptr);
+            }
+
+            auto bodyExpr = body->takeDesugaredExpr();
+            auto rescueExpr = rescue->takeDesugaredExpr();
+
+            // Create a RescueCase with empty exceptions and a rescue temp var
+            ast::RescueCase::EXCEPTION_store exceptions;
+            auto rescueTemp = nextUniqueDesugarName(core::Names::rescueTemp());
+
+            // In `rescue; ...; end`, we don't want the magic <rescueTemp> variable to look
+            // as if its loc is the entire `rescue; ...; end` span. Better to just point at
+            // the `rescue` keyword.
+            auto varLoc = (location.endPos() - location.beginPos()) > 6
```

Comment:
In a follow-up PR, can you please extract a method like `core::LocOffsets.length()`

https://github.com/search?q=repo%3Asorbet%2Fsorbet++%2FendPos%5C%28%5C%29+-+.*beginPos%5C%28%5C%29%2F&type=code